types/schematypes.d.ts

```diff
@@ -269,17 +269,17 @@ declare module 'mongoose' {
 
   namespace Schema {
     namespace Types {
-      class Array extends SchemaType implements AcceptsDiscriminator {
+      class Array<B = any> extends SchemaType<B> implements AcceptsDiscriminator<B> {
         /** This schema type's name, to defend against minifiers that mangle function names. */
         static schemaName: 'Array';
 
         static options: { castNonArrays: boolean; };
 
         discriminator<T, U>(name: string | number, schema: Schema<T, U>, value?: string): U;
-        discriminator<D>(name: string | number, schema: Schema, value?: string): Model<D>;
+        discriminator<D>(name: string | number, schema: Schema<D>, value?: string | number | ObjectId): Model<Omit<B, keyof D> & D>;
```

Comment:
I've just noticed

Is the `value` here supposed to also allow number and ObjectId? Because I've copied this line from the AcceptsDiscriminator interface and didn't noticed that it didn't allow for these types here

I _suppose_ they should because they implements AcceptsDiscriminator that allows it, but I'm not sure